resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsSnapshot.scala

```diff
@@ -86,10 +86,13 @@ object ExecutorPodsSnapshot extends Logging {
             sparkContainerStatusOpt match {
               case Some(sparkContainerStatus) =>
                 sparkContainerStatus.getState.getTerminated match {
+                  // If there is no terminated state we are running.
+                  case null => PodRunning(pod)
                   case t if t.getExitCode != 0 =>
                     PodFailed(pod)
                   case t if t.getExitCode == 0 =>
                     PodSucceeded(pod)
+                  // Fall through case
                   case _ =>
                     PodRunning(pod)
                 }
```

Comment:
What about the following?
```scala
sparkContainerStatus.getState.getTerminated match {
  case t if t != null and t.getExitCode != 0 =>
    PodFailed(pod)
  case t if t != null and t.getExitCode == 0 =>
    PodSucceeded(pod)
  case _ =>
    PodRunning(pod)
}
```

Reply:
Sure that would work to. Do you prefer this one?

Reply:
Yes, I do because it's shorter. :)

Reply:
I refactored this into using flatMaps so we can just pattern match on the result, WDYT?